interfaces/IApplication.h

```diff
@@ -23,17 +24,79 @@
 namespace WPEFramework {
 namespace Exchange {
 
-    // This interface gives direct access to an Application instance
+    // @json
     struct IApplication : virtual public Core::IUnknown {
 
         enum { ID = ID_APPLICATION };
 
-
         virtual ~IApplication() {}
 
-        virtual void Reset() = 0;
-        virtual void DeepLink(const string& deepLink) = 0;
+        enum resettype : uint8_t {
+            FACTORY,
+            CACHE,
+            CREDENTIALS,
+            RECORDINGS
+        };
+
+        enum launchpointtype : uint8_t {
+            UNDEFINED,
+            DIAL,
+            DEDICATED_BUTTON,
+            DEDICATED_ICON,
+            APPLICATION_LIST,
+            INTEGRATED_TILE,
+            SEARCH_RESULT,
+            SEARCH_CONTINUATION,
+            VOICE_CONTROL,
+            VOICE_SEARCH_RESULT,
+            VISUAL_GESTURE,
+            TOUCH_GESTURE,
+            EPG_GRID,
+            CHANNEL_NUMBER,
+            CHANNEL_ZAP,
+            CHANNEL_BAR,
+            WEB_BROWSER,
+            POWER_ON,
+            POWER_ON_FROM_DEDICATED_BUTTON,
+            SUSPENDED_POWER_ON,
+            RESTART,
+            SUSPENDED_RESTART,
+            RESUMED_FROM_SCREENSERVER,
+            RESUMED_FROM_STANDBY,
+            BANNER_AD,
+            TITLE_RECOMENDATION,
+            APPLICATION_PROMOTION
+        };
+
+        // @brief Resets application data
+        // @param type Type of reset to perform
+        virtual uint32_t Reset(const resettype type) = 0;
+
+        // @property
+        // @brief Application-specific identification string
+        // @param id Identifier string (e.g. ABCDEFG=00000005)
```

Comment:
I think the equal sign is something we would not like to advertise cause now I think you suggest that netflix should expose ESN=MOS5355500000001 and I think we should only expose MOS53555000000001. The one who uses this interface already has a callsign from where he got this interface e.g. Netflix plugin so my suggestion would be to not expose the detail of the implementation that it is an ESN, but just the ID from the Netflix plugin.

Reply:
Perhaps having Netflix ESN as example is unfortunate as it literally is "string equals number".

Reply:
Removed example